### PR TITLE
Improve API docs, including instructions on graphql_ppx_re.

### LINKED ISSUE #121: New GraphQL ppx instructions incomplete
The way to generate a `graphql_schema.json` is different on `graphql_ppx_re` so the instructions on the README right now will lead to an error when setting up a new project

We should copy the instructions from `graphql_ppx_re` and have another section with instructions if the user decides to use `graphql_ppx` only

### PR DESCRIPTION
**Pull request by parkerziegler**
Fix #121. 

This PR begins the work to improve our API documentation and clarify language around using `graphql_ppx_re`. The biggest hurdle with the new lib is the difference in how a user derives a `graphql_schema.json` file (via `graphql-cli` rather than a local Node script that ships with the lib).
We'll probably want to begin converting our examples over gradually to use the new PPX – perhaps a good issue for a first time contributor.

I also added a bit more to our `getting_started.md` to discuss `useMutation`. Recent changes to the `urql` README added a lot of good information that it may be worthwhile to piggyback off of here as well, so I'll keep working on making the docs more usable. Open to any suggestions on how to improve things.

The last change in this PR is just ensuring we're using `CombinedError.t` instead of `CombinedError.combinedError` to type that record across the lib.

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -83,8 +83,87 @@ let make = () => {
...
+  /* Pass the request to useMutation. */
+  let (_, executeMutation) = useMutation(~request, ());
+
+  <button onClick=(_e) => executeMutation()>
+      "Execute the Mutation (and Reward a Good Dog)"->React.string
+  </button>
+}
+```
+
+Great – we've successfully executed a mutation to like a dog! Existing queries that reference the mutated data will be notified that data has changed, meaning you don't need to think at all about refetching your data – it's all updated for you by `reason-urql`.
```
Comment (gugahoa) [inline docs/getting_started.md]: Oh, this is quite nice. I didn't know that it worked like this :thinking: 
How do I make sure I'm using this feature?

File: docs/getting_started.md [hunk 2]
```diff
@@ -83,8 +83,87 @@ let make = () => {
...
+
+`useMutation` returns a two-dimensional tuple, containing `(result, executeMutation)`. `result` contains the `response` variant, which allows you to pattern match on the API response from your mutation. For example, if you wanted to show different UI when your mutation was `Fetching`, or if there was an `Error(e)` you can do something like the following:
+
+```reason
+[@react.component]
+let make = (~key: string) => {
+  /* Build your request by calling .make on the graphql_ppx module, passing variables as labeled arguments. */
```
Comment (gugahoa) [inline docs/getting_started.md]: Here you mention `graphql_ppx` but in other places you removed the mention

File: docs/getting_started.md (hunk 1 above)
Comment (parkerziegler) [reply docs/getting_started.md]: It has to do with `urql`'s observable system, which is powered by [`wonka`](https://github.com/kitten/wonka). Essentially, every operation that passes through the Client can have subscribers, and this is particularly important in the case of queries that are dependent on mutation results. When a mutation is executed, `urql` will aggressively invalidate the cache for any entity that has a shared `__typename` with the mutation. This then notifies subscribers to those cache entries (really any query that requests data with the same `__typename`) that they should update: https://github.com/FormidableLabs/urql#document-caching. The first example project in this repo shows this pretty well – if you hit the button to fire `executeMutation`, the query will be notified that the cache was invalidated and pull anew from it, causing a re-render. Hope that makes sense! You don't need to do anything to opt into this, it should just happen for you.

File: docs/getting_started.md (hunk 2 above)
Comment (parkerziegler) [reply docs/getting_started.md]: Good catch, fixed here: f67eb9f5216b3f82dc4f893edfb024ccc15bffe8

**Pull request merged**